django_tables2/templatetags/django_tables2.py

```diff
@@ -206,16 +203,24 @@ def title(value):
 
     Same as Django's builtin `~django.template.defaultfilters.title` filter,
     but operates on individual words and leaves words unchanged if they already
-    have a capital letter.
+    have a capital letter or a digit. Actually Django's filter also skips
+    words with digits but only for latin letters (or at least not for
+    cyrillic ones).
     '''
-
-    def title_word(w):
-        return w if RE_UPPERCASE.search(w) else old_title(w)
-
-    return re.sub('(\S+)', lambda m: title_word(m.group(0)), value)
+    return ' '.join([
+        any([c.isupper() or c.isdigit() for c in w]) and w or old_title(w)
+        for w in value.split()
+    ])
```

Comment:
looks good!
I guess the lazyness will add a bit of overhead too, but since django is littered with lazy evaluation, it should not be too much.